app/controllers/users_controller.rb

```diff
@@ -0,0 +1,55 @@
+class UsersController < ApplicationController
+
+  get '/signup' do 
+    if !session[:user_id]
+      erb :"users/create_user"
+    else
+      redirect "/tweets"
+    end
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == "" 
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.